packages/nextjs/hooks/scaffold-eth/useFetchBlocks.ts

```diff
@@ -65,55 +60,57 @@ export const useFetchBlocks = () => {
     }
-    setIsLoading(false);
-  }, [client, currentPage]);
+  }, [currentPage]);
 
   useEffect(() => {
     fetchBlocks();
   }, [fetchBlocks]);
 
   useEffect(() => {
-    const handleNewBlock = async (newBlock: Block) => {
+    const handleNewBlock = async (newBlock: any) => {
       try {
-        if (!blocks.some(block => block.number === newBlock.number)) {
-          if (currentPage === 0) {
-            setBlocks(prevBlocks => [newBlock, ...prevBlocks.slice(0, BLOCKS_PER_PAGE - 1)]);
-
-            newBlock.transactions.forEach(tx => decodeTransactionData(tx as Transaction));
-
-            const receipts = await Promise.all(
-              newBlock.transactions.map(async tx => {
-                try {
-                  const receipt = await client.getTransactionReceipt({ hash: (tx as Transaction).hash });
-                  return { [(tx as Transaction).hash]: receipt };
-                } catch (err) {
-                  setError(err instanceof Error ? err : new Error("An error occurred."));
-                  throw err;
-                }
-              }),
+        if (currentPage === 0) {
+          if (newBlock.transactions.length > 0) {
+            const transactionsDetails = await Promise.all(
+              newBlock.transactions.map((txHash: string) => extendedClient.getTransaction({ hash: txHash as Hash })),
             );
-
-            setTransactionReceipts(prevReceipts => ({ ...prevReceipts, ...Object.assign({}, ...receipts) }));
-          }
-          if (newBlock.number) {
-            setTotalBlocks(newBlock.number);
+            newBlock.transactions = transactionsDetails;
           }
+
+          newBlock.transactions.forEach((tx: Transaction) => decodeTransactionData(tx as Transaction));
+
+          const receipts = await Promise.all(
+            newBlock.transactions.map(async (tx: Transaction) => {
+              try {
+                const receipt = await extendedClient.getTransactionReceipt({ hash: (tx as Transaction).hash });
+                return { [(tx as Transaction).hash]: receipt };
+              } catch (err) {
+                setError(err instanceof Error ? err : new Error("An error occurred fetching receipt."));
+                throw err;
+              }
+            }),
+          );
+
+          setBlocks(prevBlocks => [newBlock, ...prevBlocks.slice(0, BLOCKS_PER_PAGE - 1)]);
+          setTransactionReceipts(prevReceipts => ({ ...prevReceipts, ...Object.assign({}, ...receipts) }));
+        }
+        if (newBlock.number) {
+          setTotalBlocks(newBlock.number);
         }
       } catch (err) {
         setError(err instanceof Error ? err : new Error("An error occurred."));
       }
     };
 
-    return client.watchBlocks({ onBlock: handleNewBlock, includeTransactions: true });
-  }, [blocks, client, currentPage]);
+    return extendedClient.watchBlocks({ onBlock: handleNewBlock, includeTransactions: true });
+  }, [blocks, currentPage]);
```

Comment:
We can probably remove `blocks` from the dependency array ? since we don't to subscribe on every new block added right?